src/reference/snakemake-style-guide.rst

```diff
@@ -46,3 +46,127 @@ Avoid the ``message`` rule attribute
+your repository's top-level ``.gitignore`` file.
+
+Use Snakemake ``params:`` block to map into ``config`` dictionary
+=================================================================
+
+For example, do this:
+
+::
+
+   params:
+       name = config["name"]
+   shell:
+       "echo {params.name:q}"
+
+instead of using the ``config`` dictionary directly in the shell
+command. This has several benefits:
+
+-  Interpolation of dictionary lookups in the shell commands is
+   non-standard and confusing. (You have to use ``{config[name]}``, for
+   example. Note that the dictionary key is unquoted.)
+
+-  Param definitions can use arbitrary Python expressions, so you can do
+   more complicated things than you can with direct interpolation, such
+   as list comprehensions.
+
+-  Snakemake can automatically discover which rules have parameter
+   values that are different than the last run and show what output
+   files are affected (``--list-params-changes``).
+
+Always use quoted (:q) interpolations
+=====================================
+
+When building shell commands to run, Snakemake does not by default
+properly quote interpolated values. This works fine if the interpolated
+value doesn't contain spaces or other special shell metacharacters (like
+quotes or backslashes), but it is fragile and a time-bomb waiting to
+break on future values.
+
+Standard best practice in any language or environment is to always quote
+parameters in generated shell commands. Snakemake supports this using
+the ``:q`` modifier for interpolation:
+
+::
+
+   params:
+       file = "filename with spaces.txt"
+   shell:
+       "wc -l {params.file:q}"
+
+Not quoting these values is also a security risk.
+
+It may be tempting to make an exception for parameters with multiple
+values where you want each become a separate command-line argument, such
+as a parameter listing three filenames. In this case, however, it's
+recommended that you make the parameter a list instead of a single
+string. Snakemake will interpolate it correctly:
+
+::
+
+   params:
+       files = ["a.txt", "b.txt", "c.txt"]
+   shell:
+       "wc -l {params.files:q}"
+
+Use triple-quoted command definitions
+=====================================
+
+Using triple-quoted (``"""`` or ``'''``) command definitions make it
+much easier to build readable commands with one-option per line. It also
+avoids any nested quoting issues if you need to use literal single or
+double quotes in your command.
+
+Example:
+
+::
+
+   shell:
+       """
+       augur parse \
+           --sequences {input:q} \
+           --fields {params.fields:q} \
+           --output-sequences {output.sequences:q} \
+           --output-metadata {output.metadata:q}
+       """
+
+Always use relative paths
+=========================
+
+Relative paths (paths that don't start with ``/``) mean that anyone can
+run the build without running into portability issues caused by paths
+specific to your computer.
```

Comment:
Ah, maybe link to [Snakemake docs on relative paths](https://snakemake.readthedocs.io/en/stable/project_info/faq.html#how-does-snakemake-interpret-relative-paths) since they are interpreted differently for different directives in Snakemake.

Reply:
Done: 9414b01774c8289257dfd8477b27ac330a71b95c